RHDEVDOCS-4131 - Added 1.6 release notes (#46861)
openshift/openshift-docs: modules/gitops-release-notes-1-6-0.adoc

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

[id="gitops-release-notes-1-6-0_{context}"]
= Release notes for {gitops-title} 1.6.0

{gitops-title} 1.6.0 is now available on {product-title} 4.8, 4.9, and 4.10.

[id="new-features-1-6-0_{context}"]

> == New features

Reviewer: For IBM Z and IBM Power support please add to the new features: 

This release of {gitops-title} adds support for IBM Z and IBM Power on {product-title} 4.10.

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

[id="gitops-release-notes-1-6-0_{context}"]
= Release notes for {gitops-title} 1.6.0

{gitops-title} 1.6.0 is now available on {product-title} 4.9, 4.10 and 4.11.

[id="new-features-1-6-0_{context}"]
== New features

The current release adds the following improvements:

* With this update, the parameters of `spec.sso` field that controlled the keycloak configurations are moved to `.spec.sso.keycloak`.
The parameters of `.spec.dex` field are moved to `.spec.sso.dex`. Dex can now be enabled/disabled using `.spec.sso.provider`. The existing configuration methods will be supported alongside the new configuration methods until GitOps v1.9, when support for `.spec.dex`, `DISABLE_DEX` and `.spec.sso` fields for keycloak configuration will be removed from the operator link:https://issues.redhat.com/browse/GITOPS-1983[GITOPS-1983]

> * With this update, the Argo CD notifications controller is available as an optional workload that can be enabled/disabled through `.spec.notifications.enabled` parameter in the Argo CD CR. link:https://issues.redhat.com/browse/GITOPS-1917[GITOPS-1917]

Reviewer 2: It should be Argo CD Notifications Controller.  We should mention that Argo CD Notifications is released as Technical Preview, see the description in https://issues.redhat.com/browse/GITOPS-927

* With this update, on {product-title} 4.11, the {gitops-title} **Environments Details** page in the {gitops-title} developer perspective shows history of the successful deployments of the application environments along with links to the revision for each deployment. link:https://issues.redhat.com/browse/GITOPS-1269[GITOPS-1269]

* With this update, you can manage resources with Argo CD that are also being used as template resources or "source" by an Operator. link:https://issues.redhat.com/browse/GITOPS-982[GITOPS-982]

* This release of {gitops-title} adds support for IBM Z and IBM Power on {product-title} 4.10.

[id="fixed-issues-1-6-0_{context}"]
== Fixed issues

The following issues have been resolved in the current release:

* Before this update, `system:serviceaccount:argocd:gitops-argocd-application-controller` cannot create resource "prometheusrules" in API group `monitoring.coreos.com` in the namespace `webapps-dev`. This update fixes this issue and {gitops-title} is now able to manage all resources from the `monitoring.coreos.com` API group. link:https://issues.redhat.com/browse/GITOPS-1638[GITOPS-1638]

> * Before this update, While reconciling cluster permissions, every time the secret was getting deleted if the secret belonged to a cluster config instance. This update fixes this issue and now the `namespaces` field from the secret is deleted instead of deleting the complete secret.

Reviewer 2: This is [GITOPS-1777](https://issues.redhat.com/browse/GITOPS-1777)

* With this update, you can set the value of the `resourceTrackingMethod` to select the tracking method used by Argo CD. link:https://issues.redhat.com/browse/GITOPS-1862[GITOPS-1862]

* With this update, you can add entries to the `argocd-cm` configmap using the `extraConfig` field of Argo CD custom resource. The entries specified are reconciled to the live `config-cm` configmap without validations. link:https://issues.redhat.com/browse/GITOPS-1964[GITOPS-1964]

* With this update, on {product-title} 4.11, the {gitops-title} **Environments Details** page in the {gitops-title} developer perspective shows history of the successful deployments of the application environments along with links to the revision for each deployment. link:https://issues.redhat.com/browse/GITOPS-1269[GITOPS-1269]

* With this update, you can manage resources with Argo CD that are also being used as template resources or "source" by an Operator. link:https://issues.redhat.com/browse/GITOPS-982[GITOPS-982]

> * This release of {gitops-title} adds support for IBM Z and IBM Power on {product-title} 4.10.

Reviewer 2: IBM P/Z on all supported OCP, right?  That is OCP 4.7-4.10.

Author: @SNiemann15 please suggest

Reviewer 2: OK. I believe IBM P/Z are only tested on OCP 4.10 so the note is probably fine.

Reviewer: Yes correct Gitops support for P/Z only on OCP 4.10

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

[id="gitops-release-notes-1-6-0_{context}"]
= Release notes for {gitops-title} 1.6.0

{gitops-title} 1.6.0 is now available on {product-title} 4.9, 4.10 and 4.11.

[id="new-features-1-6-0_{context}"]
== New features

The current release adds the following improvements:

* With this update, `ApplicationSet` is now available in GA and is merged to Argo CD core.

* With this update, the parameters of `spec.sso` field that controlled the keycloak configurations are moved to `.spec.sso.keycloak`.
The parameters of `.spec.dex` field are moved to `.spec.sso.dex`. Dex can now be enabled/disabled using `.spec.sso.provider`. The existing configuration methods will be supported alongside the new configuration methods until GitOps v1.9, when support for `.spec.dex`, `DISABLE_DEX` and `.spec.sso` fields for keycloak configuration will be removed from the operator link:https://issues.redhat.com/browse/GITOPS-1983[GITOPS-1983]

> * With this update, the Argo CD Notifications Controller is available as an optional workload that can be enabled/disabled through `.spec.notifications.enabled` parameter in the Argo CD custom resource. The Argo CD Notifications Controller is available as technical preview. link:https://issues.redhat.com/browse/GITOPS-1917[GITOPS-1917]

Reviewer 2: Should "technical review" be capitalized?

Reviewer: It should be Technology Preview and I think the doc guidelines say `/snippets/technology-preview.adoc` should be included.

* With this update, the default resource exclusions are added to the `csv` for Tekton pipeline runs and task runs. Argo CD, by default prunes these resources. These resource exclusions are added to the new Argo CD instances that are created from the *OpenShift console*. If the instances are created from the CLI the resources are not added. link:https://issues.redhat.com/browse/GITOPS-1876[GITOPS-1876]

* With this update, you can set the value of the `resourceTrackingMethod` to select the tracking method used by Argo CD. link:https://issues.redhat.com/browse/GITOPS-1862[GITOPS-1862]

> * With this update, you can add entries to the `argocd-cm` configmap using the `extraConfig` field of Argo CD custom resource. The entries specified are reconciled to the live `config-cm` configmap without validations. link:https://issues.redhat.com/browse/GITOPS-1964[GITOPS-1964]

Reviewer 2: ```suggestion
* With this update, you can add entries to the `argocd-cm` configmap using the `extraConfig` field of GitOps Operator Argo CD custom resource. The entries specified are reconciled to the live `config-cm` configmap without validations. link:https://issues.redhat.com/browse/GITOPS-1964[GITOPS-1964]
```

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

[id="gitops-release-notes-1-6-0_{context}"]
= Release notes for {gitops-title} 1.6.0

{gitops-title} 1.6.0 is now available on {product-title} 4.8, 4.9, and 4.10.

[id="new-features-1-6-0_{context}"]
== New features

The current release adds the following improvements:

* With this update, `ApplicationSet` is now available in GA and is merged to Argo CD core. link:https://issues.redhat.com/browse/GITOPS-1958[GITOPS-1958]

> * With this update, the latest releases of the {gitops-title} will only be available in Latest and version based channels like, **gitops-1.6**. Users existing on the Stable channel should move to either the Latest or the version based channel for getting upgrades. link:https://issues.redhat.com/browse/GITOPS-1791[GITOPS-1791]

Reviewer 3: Please write Latest as latest and in bold

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

> 

Reviewer 4: Every file must have a content type attribute.
```suggestion
:_content-type: REFERENCE
```

* With this update, the parameters of `spec.sso` field that controlled the keycloak configurations are moved to `.spec.sso.keycloak`.
The parameters of `.spec.dex` field are moved to `.spec.sso.dex`. Dex can now be enabled/disabled using `.spec.sso.provider`. The existing configuration methods will be supported alongside the new configuration methods until GitOps v1.9, when support for `.spec.dex`, `DISABLE_DEX` and `.spec.sso` fields for keycloak configuration will be removed from the operator link:https://issues.redhat.com/browse/GITOPS-1983[GITOPS-1983]

* With this update, the Argo CD Notifications Controller is available as an optional workload that can be enabled/disabled through `.spec.notifications.enabled` parameter in the Argo CD custom resource. The Argo CD Notifications Controller is available as Technical Preview. link:https://issues.redhat.com/browse/GITOPS-1917[GITOPS-1917]

> * With this update, the default resource exclusions are added to the `csv` for Tekton pipeline runs and task runs. Argo CD, by default prunes these resources. These resource exclusions are added to the new Argo CD instances that are created from the *OpenShift console*. If the instances are created from the CLI the resources are not added. link:https://issues.redhat.com/browse/GITOPS-1876[GITOPS-1876]

Reviewer 5: We can update this to - resource exclusions for Tekton pipeline runs and tasks runs are added by default. Argo CD, by default ........ (same as above)

> * With this update, you can set the value of the `resourceTrackingMethod` to select the tracking method used by Argo CD. link:https://issues.redhat.com/browse/GITOPS-1862[GITOPS-1862]

Reviewer 4: - Label `resourceTrackingMethod`. Is it a parameter or object?: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-object-references
- I don't see it documented in previous versions of openshift-docs. Is this a new feature? So `resourceTrackingMethod` is new, not just the ability to select the tracking method. If so, how about:
```suggestion
* With this update, you can select the tracking method that by Argo CD uses by setting the `resourceTrackingMethod` parameter in <the location of this parameter>. link:https://issues.redhat.com/browse/GITOPS-1862[GITOPS-1862]
```

> * With this update, you can add entries to the `argocd-cm` configmap using the `extraConfig` field of {gitops-title} Argo CD custom resource. The entries specified are reconciled to the live `config-cm` configmap without validations. link:https://issues.redhat.com/browse/GITOPS-1964[GITOPS-1964]

Author: @gabriel-rh it should be configMap, made the change

> * With this update, you can add entries to the `argocd-cm` configmap using the `extraConfig` field of {gitops-title} Argo CD custom resource. The entries specified are reconciled to the live `config-cm` configmap without validations. link:https://issues.redhat.com/browse/GITOPS-1964[GITOPS-1964]

Reviewer 6: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-object-references

That would suggest using  "the ... ConfigMap object" or " the ... config map". But if it is a specific field in the yaml file, like below, then I guess configMap is good:

```
          volumes:
            - name: "ldap-sync-volume"
              configMap:
                name: "ldap-group-syncer"
```

* With this update, on {product-title} 4.11, the {gitops-title} **Environments Details** page in the {gitops-title} developer perspective shows history of the successful deployments of the application environments along with links to the revision for each deployment. link:https://issues.redhat.com/browse/GITOPS-1269[GITOPS-1269]

* With this update, you can manage resources with Argo CD that are also being used as template resources or "source" by an Operator. link:https://issues.redhat.com/browse/GITOPS-982[GITOPS-982]

* With this update, the Operator will now configure the Argo CD workloads with the correct permissions to satisfy the Pod Security Admission that has been enabled for Kubernetes 1.24. link:https://issues.redhat.com/browse/GITOPS-2026[GITOPS-2026]

> * With this update, Config Management Plugins 2.0 is supported. The Argo CD custom resource now allows to specify sidebar containers for the repo server. link:https://issues.redhat.com/browse/GITOPS-766[GITOPS-776]

Reviewer 6: Maybe re-write as:
 
The Argo CD custom resource can now specify sidebar containers for the repo server. 

from ISG: 

**allow** verb 
Avoid stating that inanimate objects grant abilities to people, as in "the product allows you to...". Whenever possible, use a direct, user-focused alternative such as "you can use the product to..." or "with this product, users can..."

> * With this update, Config Management Plugins 2.0 is supported. The Argo CD custom resource now allows to specify sidebar containers for the repo server. link:https://issues.redhat.com/browse/GITOPS-766[GITOPS-776]

Reviewer 4: ```suggestion
* With this update, Config Management Plugins 2.0 is supported. You can use the Argo CD custom resource to specify sidebar containers for the repo server. link:https://issues.redhat.com/browse/GITOPS-766[GITOPS-776]
```
- Avoid anthropomorphism. Rephrase as a user-focused sentence.

> * With this update, all communication between the Argo CD components and Redis cache are properly secured using modern TLS encryption. link:https://issues.redhat.com/browse/GITOPS-720[GITOPS-720]

Reviewer 6: ```suggestion
* With this update, all communication between the Argo CD components and the Redis cache are properly secured using modern TLS encryption. link:https://issues.redhat.com/browse/GITOPS-720[GITOPS-720]
```

> * This release of {gitops-title} adds support for IBM Z and IBM Power on {product-title} 4.10.

Reviewer: Please add: 

Currently, installations in restricted environments are not supported on IBM Z and IBM Power.

[id="fixed-issues-1-6-0_{context}"]
== Fixed issues

The following issues have been resolved in the current release:

> * Before this update, `system:serviceaccount:argocd:gitops-argocd-application-controller` cannot create resource "prometheusrules" in API group `monitoring.coreos.com` in the namespace `webapps-dev`. This update fixes this issue and {gitops-title} is now able to manage all resources from the `monitoring.coreos.com` API group. link:https://issues.redhat.com/browse/GITOPS-1638[GITOPS-1638]

Reviewer 6: ```suggestion
* Before this update, an error is reported stating `system:serviceaccount:argocd:gitops-argocd-application-controller` cannot create resource "prometheusrules" in API group `monitoring.coreos.com` in the namespace `webapps-dev`. This update fixes this issue and {gitops-title} is now able to manage all resources from the `monitoring.coreos.com` API group. link:https://issues.redhat.com/browse/GITOPS-1638[GITOPS-1638]
```

Author: @gabriel-rh `system:serviceaccount:argocd:gitops-argocd-application-controller` is not the error message. It is the controller. Writing this as : 
Before this update, the `system:serviceaccount:argocd:gitops-argocd-application-controller` cannot create resource "prometheusrules" in API group `monitoring.coreos.com` in the namespace `webapps-dev`. This update fixes this issue and {gitops-title} is now able to manage all resources from the `monitoring.coreos.com` API group.

> * Before this update, `system:serviceaccount:argocd:gitops-argocd-application-controller` cannot create resource "prometheusrules" in API group `monitoring.coreos.com` in the namespace `webapps-dev`. This update fixes this issue and {gitops-title} is now able to manage all resources from the `monitoring.coreos.com` API group. link:https://issues.redhat.com/browse/GITOPS-1638[GITOPS-1638]

Reviewer 4: - Label everything that appears in backticks. https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-object-references
- 
```suggestion
* Before this update, the `system:serviceaccount:argocd:gitops-argocd-application-controller` did not create a `prometheusrules` resource in the `monitoring.coreos.com` API group in the `webapps-dev` namespace. This update fixes this issue and {gitops-title} is now able to manage all resources from the `monitoring.coreos.com` API group. link:https://issues.redhat.com/browse/GITOPS-1638[GITOPS-1638]
```

> * Before this update, While reconciling cluster permissions, every time the secret was getting deleted if the secret belonged to a cluster config instance. This update fixes this issue and now the `namespaces` field from the secret is deleted instead of deleting the complete secret. link:https://issues.redhat.com/browse/GITOPS-1777[GITOPS-1777]

Reviewer 6: ```suggestion
* Before this update, when reconciling cluster permissions, the secret was always deleted if the secret belonged to a cluster config instance. This update fixes this issue and now the `namespaces` field from the secret is deleted instead of deleting the complete secret. link:https://issues.redhat.com/browse/GITOPS-1777[GITOPS-1777]
```

> * Before this update, While reconciling cluster permissions, every time the secret was getting deleted if the secret belonged to a cluster config instance. This update fixes this issue and now the `namespaces` field from the secret is deleted instead of deleting the complete secret. link:https://issues.redhat.com/browse/GITOPS-1777[GITOPS-1777]

Reviewer 4: ```suggestion
* Before this update, while reconciling cluster permissions, if a secret belonged to a cluster config instance it was deleted. This update fixes this issue. Now, the `namespaces` field from the secret is deleted instead of the secret. link:https://issues.redhat.com/browse/GITOPS-1777[GITOPS-1777]
```

> * Before this update, the Operator created the `Redis StatefulSet` with `podAffinity` rules, instead of `podAntiAffinity` rules while installing the HA variant of Argo CD through the Operator. This update fixes this issue and now the Operator creates the `Redis StatefulSet` with `podAntiAffinity` rules. link:https://issues.redhat.com/browse/GITOPS-1645[GITOPS-1645]

Reviewer 6: ```suggestion
* Before this update, the Operator created the Redis `StatefulSet` object with `podAffinity` rules, instead of `podAntiAffinity` rules, while installing the HA variant of Argo CD through the Operator. This update fixes this issue and now the Operator creates the Redis `StatefulSet` object with `podAntiAffinity` rules. link:https://issues.redhat.com/browse/GITOPS-1645[GITOPS-1645]
```

> * Before this update, the Operator created the `Redis StatefulSet` with `podAffinity` rules, instead of `podAntiAffinity` rules while installing the HA variant of Argo CD through the Operator. This update fixes this issue and now the Operator creates the `Redis StatefulSet` with `podAntiAffinity` rules. link:https://issues.redhat.com/browse/GITOPS-1645[GITOPS-1645]

Reviewer 4: ```suggestion
* Before this update, if you installed the HA variant of Argo CD through the Operator, the Operator created the `Redis StatefulSet` object with `podAffinity` rules instead of `podAntiAffinity` rules. This update fixes this issue and now the Operator creates the `Redis StatefulSet` with `podAntiAffinity` rules. link:https://issues.redhat.com/browse/GITOPS-1645[GITOPS-1645]
```

> * Before this update, Argo CD **ApplicationSet** had too many `ssh` Zombie processes. This update fixes this issue and now tini process reaper has been added into the **ApplicationSet** controller. This ensures, that SIGTERM is properly passed to the running process preventing it from being a Zombie process. link:https://issues.redhat.com/browse/GITOPS-2108[GITOPS-2108]

Reviewer 6: ```suggestion
* Before this update, Argo CD **ApplicationSet** had too many `ssh` zombie processes. This update fixes this issue and now the `tini` process reaper has been added to the **ApplicationSet** controller. This ensures that SIGTERM is properly passed to the running process, preventing it from becoming a zombie process. link:https://issues.redhat.com/browse/GITOPS-2108[GITOPS-2108]
```

> * Before this update, Argo CD **ApplicationSet** had too many `ssh` Zombie processes. This update fixes this issue and now tini process reaper has been added into the **ApplicationSet** controller. This ensures, that SIGTERM is properly passed to the running process preventing it from being a Zombie process. link:https://issues.redhat.com/browse/GITOPS-2108[GITOPS-2108]

Reviewer 4: ```suggestion
* Before this update, Argo CD **ApplicationSet** had too many `ssh` Zombie processes. This update fixes this issue: it adds tini, a simple init daemon that spawns processes and reaps zombies, to the **ApplicationSet** controller. This ensures that a `SIGTERM` signal is properly passed to the running process, preventing it from being a zombie process. link:https://issues.redhat.com/browse/GITOPS-2108[GITOPS-2108]
```

[id="known-issues-1-6-0_{context}"]
== Known issues

> *  {gitops-title} Operator can make use of RHSSO (KeyCloak) through OIDC in addition to Dex. However, with a recent security fix applied, the certificate of RHSSO cannot be validated in some scenarios. link:https://issues.redhat.com/browse/GITOPS-2212[GITOPS-2212]

Reviewer 4: ```suggestion
* {gitops-title} Operator can make use of RHSSO (KeyCloak) through OIDC in addition to Dex. However, with a recent security fix applied, the certificate of RHSSO cannot be validated in some scenarios. There is no workaround at this time. link:https://issues.redhat.com/browse/GITOPS-2212[GITOPS-2212]
```
- Mention whether there is a workaround or not per the known issue template here: https://redhat-documentation.github.io/supplementary-style-guide/#release-notes
